### PR TITLE
chore: decouple sessions from tornado

### PR DESCRIPTION
**Pull request by mscolnick**
Decoupling sessions.py from any tornado logic. This is so we can switch to another webframework much more easily

### CODE REVIEW

File: marimo/_server/model.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+class SessionMode(Enum):
+    """Session mode for a session"""
+
+    # read-write
+    EDIT = 0
+    # read-only
+    RUN = 1
+
+
+class SessionHandler(metaclass=abc.ABCMeta):
```
Comment (mscolnick) [inline marimo/_server/model.py:24]: this abstraction isn't perfect, but does remove the coupling to torando. the arguments passed should be sufficient for other connection types

File: marimo/_server/server_handler.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    def on_start(
+        self,
+        mode: SessionMode,
+        connection: Connection,
+        check_alive: Callable[[], None],
+    ) -> None:
+        self.mode = mode
+        if mode == SessionMode.RUN:
+            loop = tornado.ioloop.IOLoop.current()
+            loop.remove_handler(connection.fileno())
```
Comment (akshayka) [inline marimo/_server/server_handler.py:234]: Isn't this clean up logic, which happens in `on_stop`? Why is this here?

Comment (mscolnick) [reply marimo/_server/server_handler.py:234]: incorrectly copied from the previous code. fixed

**Review by akshayka — APPROVED** [review]
